src/coprocessor/codec/convert.rs

```diff
@@ -16,6 +16,65 @@ use std::borrow::Cow;
 
 use coprocessor::select::xeval::EvalContext;
 use super::Result;
+// `UNSPECIFIED_LENGTH` is unspecified length from FieldType
+pub const UNSPECIFIED_LENGTH: i32 = -1;
+
+pub fn truncate_str(mut s: String, flen: isize) -> String {
+    if flen != UNSPECIFIED_LENGTH as isize && s.len() > flen as usize {
+        s.truncate(flen as usize);
+        s
+    } else {
+        s
+    }
+}
+
+// `overflow` returns an overflowed error.
+#[macro_export]
+macro_rules! overflow {
+    ($val:ident, $bound:ident) => ({
+        Err(box_err!("constant {} overflows {}", $val, $bound))
+    });
+}
+
+// `convert_int_to_uint` converts an int value to an uint value.
+pub fn convert_int_to_uint(val: i64, upper_bound: u64, tp: u8) -> Result<u64> {
+    if val as u64 > upper_bound {
+        return overflow!(val, tp);
+    }
+    Ok(val as u64)
+}
+
+// `convert_uint_to_int` converts an uint value to an int value.
+pub fn convert_uint_to_int(val: u64, upper_bound: i64, tp: u8) -> Result<i64> {
+    if val > upper_bound as u64 {
+        return overflow!(val, tp);
+    }
+    Ok(val as i64)
+}
+
+pub fn convert_float_to_int(fval: f64, lower_bound: i64, upper_bound: i64, tp: u8) -> Result<i64> {
+    let val = fval.round(); // TODO
```

Comment:
Since `tidb` implements it instead of using function from `std` directly while I think we needn't. Since I'm not sure if the performance would be ok, I add a `TODO` here @BusyJay